class.jetpack.php

```diff
@@ -2300,12 +2300,29 @@ public static function get_file_data( $file, $headers ) {
 		$data = get_file_data( $file, $headers );
 
 		// Strip out any old Jetpack versions that are cluttering the option.
-		$file_data_option = array_intersect_key( (array) $file_data_option, array( JETPACK__VERSION => null ) );
+		//
+		// We maintain the data for the current version of Jetpack plus the previous version
+		// to prevent repeated DB hits on large sites hosted with multiple web servers
+		// on a single database (since all web servers might not be updated simultaneously)
+		
 		$file_data_option[ JETPACK__VERSION ][ $key ] = $data;
+		
+		if ( count( $file_data_option ) > 2 ) {
+			$count = 0;
+			krsort( $file_data_option );
```

Comment:
We have never released a '.10', and will never do so-- Jetpack doesn't practice semver